android/src/main/kotlin/com/amplitude/amplitude_flutter/AmplitudeFlutterPlugin.kt

```diff
@@ -263,236 +139,157 @@ class AmplitudeFlutterPlugin : FlutterPlugin, MethodCallHandler {
+        event.groupProperties = json.optJSONObject("group_properties")?.let {
+            it.toMutableMap()
+        } ?: null
+        event.userId = json.getString("user_id")
+        event.deviceId = json.getString("device_id")
+        event.timestamp = json.optLong("timestamp")
+        event.eventId = json.optLong("event_id")
+        event.sessionId = json.optLong("session_id")
+        event.insertId = json.optString("insert_id")
+        event.locationLat = json.optDouble("location_lat")
+        event.locationLng = json.optDouble("location_lng")
+        event.appVersion = json.getString("app_version")
+        event.versionName = json.getString("version_name")
+        event.platform = json.getString("platform")
+        event.osName = json.getString("os_name")
+        event.osVersion = json.getString("os_version")
+        event.deviceBrand = json.getString("device_brand")
+        event.deviceManufacturer = json.getString("device_manufacturer")
+        event.deviceModel = json.getString("device_model")
+        event.carrier = json.getString("carrier")
+        event.country = json.getString("country")
+        event.region = json.getString("region")
+        event.city = json.getString("city")
+        event.dma = json.getString("dma")
+        event.idfa = json.getString("idfa")
+        event.idfv = json.getString("idfv")
+        event.adid = json.getString("adid")
+        event.appSetId = json.getString("app_set_id")
+        event.androidId = json.getString("android_id")
+        event.language = json.getString("language")
+        event.library = json.getString("library")
+        event.ip = json.getString("ip")
+        event.plan = Plan(
+            plan.getString("branch"),
+            plan.getString("source"),
+            plan.getString("version"),
+            plan.getString("versionId")
+        )
+        event.ingestionMetadata = IngestionMetadata(
+            ingestionMetadata.getString("sourceName"),
+            ingestionMetadata.getString("sourceVersion")
+        )
+        event.revenue = json.optDouble("revenue")
+        event.price = json.optDouble("price")
+        event.quantity = json.optInt("quantity")
+        event.productId = json.getString("product_id")
+        event.revenueType = json.getString("revenue_type")
+        event.extra = json.optJSONObject("extra")?.let {
+            it.toMap()
+        } ?: null
+        event.partnerId = json.getString("partner_id")
+
+        return event
+    }
 
-                    // CLEARALL
-                    "\$clearAll" -> {
-                        identify.clearAll()
-                    }
-                }
+    internal fun JSONObject.toMutableMap(): MutableMap<String, Any?> {
+        val map = mutableMapOf<String, Any?>()
+        val keys = keys()
+        while (keys.hasNext()) {
+            val key = keys.next()
+            var value = get(key)
+            if (value is JSONObject) {
+                value = value.toMap()
```

Comment:
Should this be mutable too? e.g. `value.toMutableMap()`

Reply:
This function will be removed in my next PR.